addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -165,6 +170,42 @@ private void checkMetaElementsMatches(Source source) {
         }
     }
 
+    private void checkDomElementMatches(HttpMessage msg) {
+        Document doc = Jsoup.parse(msg.getResponseBody().toString());
+
+        for (Map<String, Map<String, Map<String, AppPattern>>> domSelectorMap :
+                currentApp.getDom()) {
+            for (Map.Entry<String, Map<String, Map<String, AppPattern>>> selectorMap :
+                    domSelectorMap.entrySet()) {
+                for (Map.Entry<String, Map<String, AppPattern>> nodeSelectorMap :
+                        selectorMap.getValue().entrySet()) {
+                    for (Map.Entry<String, AppPattern> value :
+                            nodeSelectorMap.getValue().entrySet()) {
+                        Elements selectedElements = doc.select(selectorMap.getKey());
+                        for (org.jsoup.nodes.Element selectedElement : selectedElements) {
+                            if (Objects.equals(value.getKey(), "text")) {
```

Comment:
Could you add a test for this case?

Reply:
`shouldMatchDomElementWithText` I guess.